packages/block-editor/src/components/block-navigation/tree.js

```diff
@@ -99,3 +105,7 @@ export default function BlockNavigationTree( {
 		</TreeGrid>
 	);
 }
+function blockNavigationTreeIsEqual( prevProps, nextProps ) {
+	return prevProps?.blocks === nextProps?.blocks;
```

Comment:
Memo'd components will still update when state changes (component state, or via hooks like our data stores). In this case checking if blocks are updated is enough to fix functionality in the navigation list modal. (The persistent list view is purely driven by state in the editor block store and not the prop).

We ignore other props like `onSelect`, since callers always return a new function.

https://github.com/WordPress/gutenberg/blob/979fb042e70dac4345d2ee8a08b1398a5b7827be/packages/block-editor/src/components/block-navigation/tree.js#L48-L53

Reply:
Wouldn't be better to use `useCallback` on the above function? There are still quite enough attributes that could change and skip the rerender.

Reply:
We could, but it's dependant on all callers to implement this correctly which makes it pretty fragile. For example the following useCallback was added without checking to see if it works. 

https://github.com/WordPress/gutenberg/blob/979fb042e70dac4345d2ee8a08b1398a5b7827be/packages/block-editor/src/components/block-navigation/tree.js#L48-L53

It doesn't in the post editor, as onSelect is always a new function.

https://github.com/WordPress/gutenberg/blob/979fb042e70dac4345d2ee8a08b1398a5b7827be/packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js#L29-L32

Reply:
I agree, but do you think checking only the `blocks` is enough for `blockNavigationTreeIsEqual`?

Reply:
Thanks for looking @ntsekouras I'll circle back to this and audit other props after we see if folks like potential component memos in https://github.com/WordPress/gutenberg/pull/32469